modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts

```diff
@@ -2,18 +2,33 @@
  * Zero Knowledge Range Proofs as described in (Two-party generation of DSA signatures)[1].
  * [1]: https://reitermk.github.io/papers/2004/IJIS.pdf
  */
-import { createHash } from 'crypto';
+import { createHash, generatePrime, GeneratePrimeOptions } from 'crypto';
 import BaseCurve from '../../curves';
 import { PublicKey } from 'paillier-bigint';
-import { bitLength, prime, randBits, randBetween } from 'bigint-crypto-utils';
+import { bitLength, randBits, randBetween } from 'bigint-crypto-utils';
 import { gcd, modPow } from 'bigint-mod-arith';
 import { NTilde, RangeProof, RangeProofWithCheck } from './types';
 import { bigIntFromBufferBE, bigIntToBufferBE } from '../../util';
 
+export async function generateSafePrime(bitlength: number): Promise<bigint> {
+  return new Promise<bigint>((resolve) => {
+    const options: GeneratePrimeOptions = {
+      safe: true,
+      bigint: true,
+    };
+    generatePrime(bitlength, options, (err, prime) => {
+      resolve(prime as bigint);
```

Comment:
We can also reject the err incase an error appears

Reply:
Yeah that's fair. I'll add that before I try and wrangle approvals